x509/revocation/crl/crl.go

```diff
@@ -12,46 +12,49 @@ import (
 // RevocationReasonCode - status codes that explain revocation reason see RFC 5280, Section 5.3.1
 type RevocationReasonCode int
 
-const (
-	Unspecified          RevocationReasonCode = 0
-	KeyCompromise        RevocationReasonCode = 1
-	CACompromise         RevocationReasonCode = 2
-	AffiliationChanged   RevocationReasonCode = 3
-	Superseded           RevocationReasonCode = 4
-	CessationOfOperation RevocationReasonCode = 5
-	CertificateHold      RevocationReasonCode = 6
+var reasonCodeNames map[RevocationReasonCode]string
+
+func init() {
+	reasonCodeNames[0] = "unspecified"
+	reasonCodeNames[1] = "keyCompromise"
+	reasonCodeNames[2] = "cACompromise"
+	reasonCodeNames[3] = "affiliationChanged"
+	reasonCodeNames[4] = "superseded"
+	reasonCodeNames[5] = "cessationOfOperation"
+	reasonCodeNames[6] = "certificateHold"
 	// STATUS CODE 7 IS NOT USED
-	RemoveFromCRL      RevocationReasonCode = 8
-	PrivilegeWithdrawn RevocationReasonCode = 9
-	AACompromise       RevocationReasonCode = 10
-)
+	reasonCodeNames[8] = "removeFromCRL"
+	reasonCodeNames[9] = "privilegeWithdrawn"
+	reasonCodeNames[10] = "aACompromise"
+}
+
+// MarshalJSON implements the json.Marshler interface
+func (code *RevocationReasonCode) MarshalJSON() ([]byte, error) {
+	aux := struct {
+		Code  int
+		Value string
+	}{
+		Code:  int(*code),
+		Value: code.String(),
+	}
+	return json.Marshal(&aux)
+}
 
-func (r RevocationReasonCode) String() string {
-	switch r {
-	case Unspecified:
-		return "unspecified"
-	case KeyCompromise:
-		return "keyCompromise"
-	case CACompromise:
-		return "cACompromise"
-	case AffiliationChanged:
-		return "affiliationChanged"
-	case Superseded:
-		return "superseded"
-	case CessationOfOperation:
-		return "cessationOfOperation"
-	case CertificateHold:
-		return "certificateHold"
-		// STATUS CODE 7 IS NOT USED
-	case RemoveFromCRL:
-		return "removeFromCRL"
-	case PrivilegeWithdrawn:
-		return "privilegeWithdrawn"
-	case AACompromise:
-		return "aACompromise"
-	default:
-		return "Unknown revocation reason code: " + strconv.Itoa(int(r))
+// UnmarshalJSON implements the json.Unmarshaler interface
+func (code *RevocationReasonCode) UnmarshalJSON(b []byte) error {
+	aux := struct {
+		Code  int
```

Comment:
Should I move my definitions to https://github.com/zmap/zcrypto/blob/master/zcrypto_schemas/zcrypto.py? Or was that just an example of how types are defined?

Reply:
The way it works today, you would update the schema.py to match the JSON emitted by your go code -- Alex is working on a way to reduce this duplication, but I don't think it's been finalized yet.

Currently the schema updates are only needed if your results are going into BigTable / ElasticSearch, since that's what we use to generate the schema.

Reply:
Ok, I don't think that these results are going into BigTable just yet. Maybe I can plan on updating schema.py when it's time to start adding these results to BigTable?

Reply:
Yeah, I don't see any reason the schema needs to be updated immediately.